File: CrossSetaDeduplicator/src/CrossSetaWeb/Services/UserService.cs
using System;
using System.Security.Cryptography;
using System.Text;
using CrossSetaWeb.DataAccess;
using CrossSetaWeb.Models;

namespace CrossSetaWeb.Services
{
    public class UserService : IUserService
    {
        private readonly IDatabaseHelper _dbHelper;

        public UserService(IDatabaseHelper dbHelper)
        {
            _dbHelper = dbHelper;
        }

        public void RegisterUser(UserModel user, string password)
        {
            if (string.IsNullOrWhiteSpace(user.UserName) || string.IsNullOrWhiteSpace(password))
            {
                throw new ArgumentException("Username and Password are required.");
            }

            user.PasswordHash = HashPassword(password);
            _dbHelper.InsertUser(user);
        }

        private string HashPassword(string password)
        {
            using (SHA256 sha256 = SHA256.Create())
            {
                byte[] bytes = sha256.ComputeHash(Encoding.UTF8.GetBytes(password));
                StringBuilder builder = new StringBuilder();
                foreach (byte b in bytes)
                {
                    builder.Append(b.ToString("x2"));
                }
                return builder.ToString();
            }
        }
    }
}

File: CrossSetaDeduplicator/src/CrossSetaWeb/CrossSetaLogic/Services/UserService.vb
Imports System
Imports System.Security.Cryptography
Imports System.Text
Imports CrossSetaLogic.DataAccess
Imports CrossSetaLogic.Models

Namespace Services
    Public Class UserService
        Implements IUserService

        Private ReadOnly _dbHelper As IDatabaseHelper

        Public Sub New(dbHelper As IDatabaseHelper)
            _dbHelper = dbHelper
        End Sub

        Public Sub RegisterUser(user As UserModel, password As String) Implements IUserService.RegisterUser
            If String.IsNullOrWhiteSpace(user.UserName) OrElse String.IsNullOrWhiteSpace(password) Then
                Throw New ArgumentException("Username and Password are required.")
            End If

            user.PasswordHash = HashPassword(password)
            _dbHelper.InsertUser(user)
        End Sub

        Private Function HashPassword(password As String) As String
            Using sha256 As SHA256 = SHA256.Create()
                Dim bytes As Byte() = sha256.ComputeHash(Encoding.UTF8.GetBytes(password))
                Dim builder As New StringBuilder()
                For Each b As Byte In bytes
                    builder.Append(b.ToString("x2"))
                Next
                Return builder.ToString()
            End Using
        End Function
    End Class
End Namespace
